File: hello-csharp/src/Server/Memory/StreamContext.cs
// Stream context for managing active audio streams.
// Contains stream metadata and cache file handle.
// Matches Python StreamContext and Java StreamContext functionality.

using System;
using System.Threading;

namespace AudioStreamServer.Memory;

/// <summary>
/// Stream status enumeration
/// </summary>
public enum StreamStatus
{
    Uploading,
    Ready,
    Error
}

/// <summary>
/// Stream context containing metadata and state for a single stream.
/// </summary>
public class StreamContext
{
    public string StreamId { get; private set; }
    public string CachePath { get; set; }
    public MemoryMappedCache? MmapFile { get; set; }
    public long CurrentOffset { get; set; }
    public long TotalSize { get; set; }
    public DateTime CreatedAt { get; private set; }
    public DateTime LastAccessedAt { get; private set; }
    public StreamStatus Status { get; set; }
    
    /// <summary>
    /// Lock object for thread-safe access to this stream context
    /// </summary>
    public object Lock { get; } = new object();

    /// <summary>
    /// Create a new StreamContext.
    /// </summary>
    public StreamContext(string streamId, string cachePath = "")
    {
        StreamId = streamId;
        CachePath = cachePath;
        MmapFile = null;
        CurrentOffset = 0;
        TotalSize = 0;
        var now = DateTime.UtcNow;
        CreatedAt = now;
        LastAccessedAt = now;
        Status = StreamStatus.Uploading;
    }

    /// <summary>
    /// Update last accessed timestamp
    /// </summary>
    public void UpdateAccessTime()
    {
        LastAccessedAt = DateTime.UtcNow;
    }
}

File: hello-dart/lib/server/memory/stream_context.dart
// Stream context for managing active audio streams.
// Contains stream metadata and cache file handle.
// Matches Python StreamContext and Java StreamContext functionality.

/// Stream status enumeration
enum StreamStatus {
  uploading,
  ready,
  error,
}

/// Stream context containing metadata and state for a single stream.
class StreamContext {
  final String streamId;
  String cachePath;
  dynamic mmapFile;
  int currentOffset;
  int totalSize;
  final DateTime createdAt;
  DateTime lastAccessedAt;
  StreamStatus status;

  /// Create a new StreamContext.
  StreamContext(this.streamId, [this.cachePath = ''])
      : mmapFile = null,
        currentOffset = 0,
        totalSize = 0,
        createdAt = DateTime.now(),
        lastAccessedAt = DateTime.now(),
        status = StreamStatus.uploading;

  /// Update last accessed timestamp
  void updateAccessTime() {
    lastAccessedAt = DateTime.now();
  }
}
